libcxx/test/benchmarks/format/write_string_comparison.bench.cpp

```diff
@@ -0,0 +1,220 @@
+                      std::string s;                                                                                   \
+                      s.resize(10'000);                                                                                \
+                      return s;                                                                                        \
+                    }));                                                                                               \
+  BENCHMARK_CAPTURE(BM_format_to_iterator, <std::vector> name, variable, ([] {                                         \
+                      std::vector<char> v;                                                                             \
+                      v.resize(10'000);                                                                                \
+                      return v;                                                                                        \
+                    }));                                                                                               \
+  BENCHMARK_CAPTURE(BM_format_to_iterator, <std::deque> name, variable, ([] {                                          \
+                      std::deque<char> d;                                                                              \
+                      d.resize(10'000);                                                                                \
+                      return d;                                                                                        \
+                    }));                                                                                               \
+                                                                                                                       \
+  /* */
+
+// Verify these types have an iterator that format has optimizations for.
+LIBCPP_STATIC_ASSERT(std::same_as<std::array<char, 1>::iterator, char*> || // the type depends on an ABI flag
+                     std::same_as<std::array<char, 1>::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(std::same_as<std::string::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(std::same_as<std::vector<char>::iterator, std::__wrap_iter<char*>>);
+
+// Verify these types have an iterator that format does not optimize for
+LIBCPP_STATIC_ASSERT(!std::same_as<std::deque<char>::iterator, char*> &&
+                     !std::same_as<std::deque<char>::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(!std::same_as<std::list<char>::iterator, char*> &&
+                     !std::same_as<std::list<char>::iterator, std::__wrap_iter<char*>>);
+
+BENCHMARK_CAPTURE(BM_sprintf, C string len = 6, c_string_6_characters);
+FORMAT_BENCHMARKS(C string len = 6, c_string_6_characters)
+FORMAT_BENCHMARKS(string len = 6, string_6_characters)
+FORMAT_BENCHMARKS(string_view len = 6, string_view_6_characters)
```

Comment:
I find the use of `=` in these names to be confusing. It almost looks like a variable definition but in reality it's not. IMO a title like `C_string_len_6` would be less confusing (also less cute, but that's not as valuable as clarity).